pkg/v1/daemon/write.go

```diff
@@ -54,15 +54,15 @@ func Tag(src, dest name.Tag) error {
 }
 
 // Write saves the image into the daemon as the given tag.
-func Write(tag name.Tag, img v1.Image) (string, error) {
+func Write(tag name.Tag, img v1.Image, opts ...tarball.WriteOption) (string, error) {
```

Comment:
It's a bit weird for `daemon.Write` to take a `tarball.Options`, it leaks a bit of the underlying abstraction.

Would it make sense to define a daemon Option for progress and convert it internally into the tarball Option?

Reply:
I think that remote.Write has the same problem, due to WriteOptions being added on the wrong level

Reply:
But the right thing to do would probably be to add a daemon.WithProgress, instead of tarball.WithProgress

Reply:
Yeah exactly. https://github.com/google/go-containerregistry/pull/967 defines a new `remote.WithProgress`. In the case of `daemon`, it might end up using `tarball.WithProgress` underneath, but it feels weird to have that leak into its API.

There isn't a `daemon.WriteOption` type yet, but I think we could add one, and a `daemon.WithProgress` that implements it, that converts to a `tarball.WriteOption` internally.

@jonjohnsonjr wdyt?

Reply:
Otherwise it seems to be working, of course we might end up using tarball anyway for Podman support.

Reply:
I've stomped on this PR a little bit by [making everything just `daemon.Option`](https://github.com/google/go-containerregistry/pull/989) (sorry), but it shouldn't be too hard to do this now by just translating the `daemon.WithProgress` into a `tarball.WithProgress`.

Reply:
I had to fork everything anyway, to make a `podman` matching the `daemon` in the API.